### PR TITLE
Issue #13345: Enabled UnnecessarySemicolonAfterTypeMemberDeclarationC…

### PR DESCRIPTION
**Pull request by kkoutsilis**
Part of #13345 
Enabled UnnecessarySemicolonAfterTypeMemberDeclarationCheckExamplesTest

### CODE REVIEW

**Comment by romani** [comment]
Github, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/Example1.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class Example1 {
+  ; // violation, standalone semicolon
+  {}; // violation, extra semicolon after init block
+  static {}; // violation, extra semicolon after static init block
+  Example1() {}; // violation, extra semicolon after constructor definition
+  void method() {}; // violation, extra semicolon after method definition
+  int field = 10;; // violation, extra semicolon after field declaration
+
+  {
+    ; // no violation, it is empty statement inside init block
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/Example1.java:20]: `// ok, it is empty statement inside init block`

please apply the same change to other "no violation" lines.
it is better be have "ok" and "violation" as very different words, as some people do not read "no".

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/Example1.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  <module name="TreeWalker">
+    <module name="UnnecessarySemicolonAfterTypeMemberDeclaration"/>
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessarysemicolonaftertypememberdeclaration;
+
+// xdoc section -- start
+class Example1 {
+  ; // violation, standalone semicolon
```
[the remark this answers is not in the document]
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/Example1.java:12]: yes, we need `// violation, '{{actual message}}'`
it helps users a lot.
if you do one more PR on more complicated Check you will see how this make huge difference.

[the remark this answers is not in the document]
Comment (kkoutsilis) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/Example1.java:12]: Will update shortly, thanks for clarifying!

[the remark this answers is not in the document]
Comment (kkoutsilis) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/Example1.java]: Addressed in https://github.com/checkstyle/checkstyle/pull/15736/commits/8c114736d158ff7af5664d979deb81a690cb76b0

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/Example1.java (hunk 1 above)
Comment (kkoutsilis) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypememberdeclaration/Example1.java]: Addressed in https://github.com/checkstyle/checkstyle/pull/15736/commits/8c114736d158ff7af5664d979deb81a690cb76b0

**Review by romani — APPROVED** [review]
good!!

but please do single commit.
please read and watch videos on how to deal with commits - https://checkstyle.org/beginning_development.html#Starting_Development

**Comment by kkoutsilis** [comment]
> good!!
> 
> but please do single commit. please read and watch videos on how to deal with commits - https://checkstyle.org/beginning_development.html#Starting_Development

Done

**Review by romani — CHANGES_REQUESTED** [review]
we need single commit.

**Review by romani — APPROVED** [review]
ok to merge if CI pass

**Comment by romani** [comment]
Github, generate website

**Comment by romani** [comment]
@kkoutsilis , please do more contribution, we appreciate your help.

**Comment by kkoutsilis** [comment]
> @kkoutsilis , please do more contribution, we appreciate your help.

Will do :)

**Pull request merged**